src/features/routing/routing.ts

```diff
@@ -14,17 +14,44 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { injectable } from "inversify";
-import { center, maxDistance, Point, euclideanDistance, linear } from "../../utils/geometry";
-import { Routable } from "./model";
+import { Point } from "../../utils/geometry";
+import { SRoutableElement, SConnectableElement } from "./model";
+import { InstanceRegistry } from "../../utils/registry";
+import { PolylineEdgeRouter } from "./polyline-edge-router";
+import { injectable, multiInject } from "inversify";
+import { ResolvedHandleMove } from "../move/move";
+import { SRoutingHandle } from "../routing/model";
+import { TYPES } from "../../base/types";
 
 export interface RoutedPoint extends Point {
     kind: 'source' | 'target' | 'linear'
     pointIndex?: number
 }
 
+export interface EdgeSnapshot {
+    routingHandles: SRoutingHandle[]
+    routingPoints: Point[]
+    router: IEdgeRouter
+    source?: SConnectableElement
+    target?: SConnectableElement
+}
+
+export interface EdgeMemento {
+    edge: SRoutableElement
+    before: EdgeSnapshot
+    after: EdgeSnapshot
+}
+
 export interface IEdgeRouter {
-    route(edge: Routable): RoutedPoint[]
+
+    readonly kind: string;
+
+    /**
+     * Calculates the route of the given edge.
+     *
+     * @param edge
```

Comment:
Minor: remove undocumented `@param` tag. Same for other methods in this interface.

Reply:
Done